src/packages/migrate/src/MigrateEngine.ts

```diff
@@ -221,40 +223,44 @@ export class MigrateEngine {
 
         this.child.on('error', (err) => {
           console.error('[migration-engine] error: %s', err)
-          reject(err)
           this.rejectAll(err)
+          reject(err)
         })
 
-        this.child.on('exit', (code) => {
-          const messages = this.messages.join('\n')
-          let err: RustPanic | Error | undefined
-          if (
-            code !== MigrateEngineExitCode.Success ||
-            messages.includes('panicking')
-          ) {
-            let errorMessage =
-              chalk.red.bold('Error in migration engine: ') + messages
-            if (this.lastError && code === MigrateEngineExitCode.Panic) {
-              errorMessage = serializePanic(this.lastError)
-              err = new RustPanic(
-                errorMessage,
-                this.lastError.message,
-                this.lastRequest,
-                ErrorArea.LIFT_CLI,
-                this.schemaPath,
-              )
-            } else if (code === MigrateEngineExitCode.Panic) {
-              err = new RustPanic(
-                errorMessage,
-                messages,
+        this.child.on('exit', (code: number | null): void => {
+          const exitWithErr = (err: RustPanic | Error): void => {
+            this.rejectAll(err)
+            reject(err)
+          }
+          const engineMessage =
+            this.lastError?.message || this.messages.join('\n')
```

Comment:
out of context bikeshedding, but I find the line length prettier is configured with on this repo shockingly short

Reply:
We agree, we were thinking switching to 100 or so soon!